api/src/main/java/jakarta/enterprise/inject/spi/BeanContainer.java

```diff
@@ -251,4 +251,42 @@ public interface BeanContainer {
      * @since 2.0
      */
     Instance<Object> createInstance();
+
+    /**
+     * Returns true if a bean with given bean types and qualifiers would be assignable
+     * to an injection point with given required type and required qualifiers, false otherwise.
+     * In other words, this method provides access to the typesafe resolution rules - the process of matching a bean to an injection point.
```

Comment:
In my previous wording, I tried to _not_ make an impression that this method represents the entirety of typesafe resolution rules, because there's more to it than just assignability. The current wording is a little more ambiguous.

But maybe that's not a big deal and this sentence could be dropped entirely?

(Same for observers, although unfortunately the spec doesn't have a nice single term for what `isMatchingEvent()` does, unlike `isMatchingBean()`, which just implements _assignability_.)

Reply:
Hmm, I was trying to explain it plainly but you're right that it introduces ambiguity.
Will remove the sentence.